src/rpc_client/mod.rs

```diff
@@ -510,8 +548,18 @@ pub enum MultiCallError<T> {
     InconsistentResults(MultiCallResults<T>),
 }
 
-impl<T: Debug + PartialEq> MultiCallResults<T> {
-    pub fn reduce_with_equality(self) -> Result<T, MultiCallError<T>> {
+impl<T: Debug + PartialEq + Clone + Serialize> MultiCallResults<T> {
+    pub fn reduce(self, strategy: ConsensusStrategy) -> Result<T, MultiCallError<T>> {
+        match strategy {
+            ConsensusStrategy::Equality => self.reduce_with_equality(),
+            ConsensusStrategy::Threshold {
+                num_providers: _,
```

Comment:
Why is the number of providers ignored here?

Reply:
That number is only required to choose the providers but once the providers are chosen, we don't need it (or I'm not sure what you would want to check with it). We now only need to make sure that we have at least `min_num_ok` consistent ok responses. Does-it make sense?

Reply:
Yes, that makes sense. Thanks!